### DISCUSSION TITLE
Need to exclude a file from merging to main branch

### DISCUSSION THREAD

**Discussion by speter04** [community/community#29112] [Pull Requests]
Hi Team,

I have a use case as given below.
In my main branch i have 2 files one is fileA and another is file_template. 
A developer can cut out a branch from main branch and update changes to both the files.
he can raise a pull request to main branch for merging the code to main branch, at that time the template file file_template needs to be excluded as its a template file and it need not be merged to main branch(its a static file in main and differ from branch to branch, their is no need to merge file_template to main branch ).

Is their any direct method we can achieve this?

**Comment by airtower-luna** [comment]
No, nothing easy. You could of course cherry-pick commits without changes to the template, but that would cause a mess if you then later merge `main` into a developer branch, because to Git a cherry-pick is a new commit.

An important question here is: What is the purpose of that `file_template`, and the files created from it? For example, if the template is a configuration template that needs to be adjusted to the developer's environment it should probably be copied to a new file that doesn't get committed. In other cases committing it under a different name might work. But what's good depends on how the file is used.
